Branch/refactor

FasterSpeeding opened the pull request.

Commit:
Unload plugins + added relevant owner commands and added keyboard Interrupt catch.
Switched to more disco-interdependent custom prefix command system
Switched to metadata system for help embed categories.
Switch from .format to f string
Shifted some dependency on thread to disco built in schedules.
Made fmPlugin Cache dependent on pre-defined session

Commit:
Shift to base.query, enabled cascade, added some commands + several fixes

FasterSpeeding committed:
Update README.md

FasterSpeeding committed:
Update README.md

Commit:
refactored embed.py
Added command category searching for help command, and added support for command groups.
Shifted to pulling artwork for artists and releases from discogs to comply with new last.fm api changes
Added new module check system for required config entries.
Laid out foundation for discogs module

Commit:
Merge branch 'branch/refactor' of https://github.com/LMByrne/Discord.FM into branch/refactor

Lyyfe commented on README.md:
```diff
@@ -1,37 +1,37 @@
+This bot is designed primarily with Python 3.6 and a Linux environment in mind, and may not function properly in other conditions.
 
-To install the neccessary Python modules use the following command in the bot's folder: 
+To install the necessary Python modules, use the following command in the bot's folder (with `sudo` not being required in all environments: 
```
close the bracket

Lyyfe commented on README.md:
```diff
@@ -1,37 +1,37 @@
-``pip3 install -r requirements.txt``
+``sudo pip3 install -r requirements.txt``
 
 In-order to enable voice functionality, you need to install ffmpeg3 and keep youtube-dl up-to-date for voice functionality to work using:
```
either the start or the end of that line is redundant, "In-order to enable voice functionality"; "for voice functionality to work"

Commit:
PEP8

Lyyfe commented on README.md:
```diff
@@ -1,37 +1,37 @@
 
 ## Setup
 
 ### Discord access
```
possessive 'its'
"take it's token" to "take its token"
maybe idk you got a higher english grade

Lyyfe commented on README.md:
```diff
@@ -42,7 +42,8 @@ In-order enable the relevant API functions, you will have to setup accounts and
 ## Optional
 
-Whilest this bot will default to making and running off an sqlite database (``data/sql_database.db``), you can hock this bot up to an SQL server by adding the following to config.json with your own data inserted:
+Whilst this bot will default to running off an automatically generated local SQLite database (``data/database.db``), you can hook it up to an SQL server by adding the example seen bellow to config.json with your own data inserted.
```
"bellow"REEEEEEEEE

Lyyfe commented on bot/base.py:
```diff
@@ -5,24 +5,31 @@
 import yaml
```
`from yaml import safe_load`
then replace `yaml.safe_load` with `safe_load`

Commit:
Enabled escape_codeblocks in multiple uses of sanitize
Switched to a persistent Song_Lyrics instance.
Cleaned up spotify oauth expire check.

Commit:
Improved on regex based redacting private data from certain outputs.
Removed instances of rededining built-in functions.
Removed wildcard import in sql.

pep8speaks commented:
Hello @LMByrne! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:





* In the file [`bot/disco/api.py`](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/api.py):

> [Line 64:1](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/api.py#L64): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 143:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/api.py#L143): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> [Line 234:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/api.py#L234): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (106 > 79 characters)

* In the file [`bot/disco/core.py`](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/core.py):

> [Line 235:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/core.py#L235): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 363:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/core.py#L363): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 452:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/core.py#L452): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (90 > 79 characters)
> [Line 472:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/core.py#L472): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (93 > 79 characters)

* In the file [`bot/disco/fm.py`](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/fm.py):

> [Line 75:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/fm.py#L75): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 139:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/fm.py#L139): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)
> [Line 191:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/fm.py#L191): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 330:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/fm.py#L330): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 366:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/fm.py#L366): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)
> [Line 420:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/fm.py#L420): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 475:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/fm.py#L475): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 530:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/fm.py#L530): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 574:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/fm.py#L574): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 617:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/fm.py#L617): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)
> [Line 661:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/fm.py#L661): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 748:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/fm.py#L748): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (101 > 79 characters)
> [Line 788:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/fm.py#L788): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 1250:29](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/fm.py#L1250): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent

* In the file [`bot/disco/superuser.py`](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/superuser.py):

> [Line 40:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/superuser.py#L40): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 62:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/superuser.py#L62): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)
> [Line 88:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/superuser.py#L88): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (105 > 79 characters)
> [Line 102:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/superuser.py#L102): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (105 > 79 characters)
> [Line 114:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/superuser.py#L114): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (98 > 79 characters)
> [Line 126:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/superuser.py#L126): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 136:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/superuser.py#L136): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 79 characters)

* In the file [`bot/disco/voice.py`](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/voice.py):

> [Line 220:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/voice.py#L220): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)
> [Line 331:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/voice.py#L331): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (92 > 79 characters)
> [Line 519:80](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/disco/voice.py#L519): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (94 > 79 characters)



* In the file [`bot/util/misc.py`](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/util/misc.py):

> [Line 53:13](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/util/misc.py#L53): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 54:13](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/util/misc.py#L54): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 55:13](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/util/misc.py#L55): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 56:13](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/util/misc.py#L56): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 57:13](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/util/misc.py#L57): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 58:13](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/util/misc.py#L58): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 59:13](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/util/misc.py#L59): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 60:13](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/util/misc.py#L60): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 61:13](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/util/misc.py#L61): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 62:13](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/util/misc.py#L62): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 63:13](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/util/misc.py#L63): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 72:1](https://github.com/LMByrne/Discord.FM/blob/4a34e06f84de5b85d827c9f58b1edf4a582edbb5/bot/util/misc.py#L72): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1

Commit:
Style and complexity 1

Commit:
2

Commit:
Remove unused variables and function imports.
Implement static methods where suitable.
Avoid dangerous default to dict.
Remove run.sh
Added to README.md

Commit:
Fix post

Commit:
Add pip path in run.py
Add some more static methods
Fix lyrics_limit handling
